rllib/offline/tests/test_dataset_writer.py

```diff
@@ -0,0 +1,71 @@
+import os
+import tempfile
+import unittest
+
+import itertools
+import boto3
+from moto import mock_s3
+from ray.rllib.offline.io_context import IOContext
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.offline.dataset_writer import DatasetWriter
+
+
+class TestDatasetWriter(unittest.TestCase):
+    def setUp(self):
+        self.ioctx = IOContext()
+        self.sample_batch = SampleBatch({"obs": [0, 1], "new_obs": [1, 2]})
+
+    def test_writer_local(self):
+        formats = ["json", "parquet"]
+        is_multiagents = [False, True]
+        for format, is_multiagent in itertools.product(formats, is_multiagents):
+            with tempfile.TemporaryDirectory() as tmpdir:
+                self.ioctx.config["output_config"] = {"format": format, "path": tmpdir}
+
+                with DatasetWriter(self.ioctx) as writer:
+                    if is_multiagent:
+                        writer.write(self.sample_batch.as_multi_agent())
+                    else:
+                        writer.write(self.sample_batch)
+
+                # check if there is at least one json file in the directory
+                self.assertTrue(any([
+                    filename.endswith(f".{format}")
+                    for filename in os.listdir(tmpdir)
+                ]))
+    @mock_s3
+    def test_writer_remote(self):
```

Comment:
@gjoliver @sven1977 @matthewdeng How do u test writing into s3 without requiring credentials? The DatasetWriter is supposed to use ds.write_xxx api. But I don't know how to test our dataset_writer with it. Any ideas?

Reply:
isn't this a mock_s3 bucket? can you just make it public?

Reply:
the mock_s3 thing was supposed to work, but I guess it creates the bucket within the context of the current python process. So if ds.write_json() creates a new process for writing then it won't have the reference to the bucket. I am not sure if this is happening tho, just a strong hypothesis I have.

Reply:
```
OSError: When testing for existence of bucket 'test-bucket': AWS Error [code 15]: No response body.
```

This is the error message on CI right now.